### DISCUSSION TITLE
StreamingResponse and async def with infinite streaming expected behavior

### DISCUSSION THREAD

**Discussion by mile95** [fastapi/fastapi#6829] [Questions] [answered]
I modified [this ](https://fastapi.tiangolo.com/advanced/custom-response/?h=stream#streamingresponse) StreamingResponse example code from the docs to use an infinite stream instead of a finite one. 

```Python
import asyncio
from fastapi import FastAPI
from fastapi.responses import StreamingResponse

app = FastAPI()


async def fake_video_streamer():
    while True:
        await asyncio.sleep(0.01)
        yield b"some fake video bytes"


@app.get("/")
async def main():
    return StreamingResponse(fake_video_streamer())


if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8000)

````
### Description

* Open the browser and call the endpoint `/`.  Repeat this 7 times so that you have 7 tabs open with the endpoint `/`.
* The last tab (the 7th) won´t load until you close one of the other tabs.
* But I expected it to load the streaming data since the `entrypoint ` and `fake_video_streamer()` are `async.`

It feels like we exhaust the thread pool (containing 6? threads ), but that should not be the case since we have defined the entrypoint async and should not involve a thread pool. 

When removing the `await asyncio.sleep(0.01)` in the async generator, we get expected behavior, namely some blocking. If running the code below where `await asyncio.sleep(0.01)` is removed and we try to reach `/` on two tabs, only the first one will succeed since we are blocking. 

```Python
import asyncio
from fastapi import FastAPI
from fastapi.responses import StreamingResponse

app = FastAPI()


async def fake_video_streamer():
    while True:
        yield b"some fake video bytes"


@app.get("/")
async def main():
    return StreamingResponse(fake_video_streamer())


if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8000)

````

I do not understand why the first scenario seems to run synchronously while scenario 2 seems to run run asynchronously. What should be the expected behavior in the first scenario? Shouldn't we be able to open multiple tabs before facing any issues (more than 6 at least)?

### Environment

* OS: Linux (wsl2)
* FastAPI Version [0.63.0]:
* Python version: 3.8.2

**Comment by ycd** [comment]
HAHAHAHA it exactly happened at the seventh tab LOL, but when i opened more than +10 CLI sessions it worked fine, so i think this is some kind of limitation of the Browser.

**Comment by falkben** [comment]
I couldn't even reproduce the 7th tab thing (Firefox). Both versions w/ & w/o the sleep worked fine for me.

**Comment by mile95** [comment, marked as answer]
Then I at least learned something new! 

Thanks, @ycd  and, @falkben  for looking into it. It is browser related, closing!

**Comment by tiangolo** [comment]
Thanks for the help here @ycd and @falkben ! 👏  🙇 

Thanks for reporting back and closing the issue @mile95  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
